### ISSUE TITLE
React Native Sentry FullStory Integration

### ISSUE THREAD

**Issue by TannerLessen** [getsentry/sentry-react-native#1966]
Any plans to support a React Native FullStory integration akin to https://github.com/getsentry/sentry-fullstory ?

**Comment by bruno-garcia** [comment]
Nothing on the roadmap. But we'll keep this feature request open here and track demand by watching 👍  on the issue

**Comment by TannerLessen** [comment]
Appreciate the quick response!

**Comment by andaydev** [comment]
Is there an easy way for us to manually search for the session in Fullstory? For example, does Sentry/Fullstory capture a session id that can be referenced in both?

**Comment by marandaneto** [comment]
@kamilogorek or @AbhiPrasad do you know how this integration work on the browser SDK? what would we need to do to integrate that with the RN SDK? if it's just a session id set in the event context, that would be easy, but I believe it's not that.

**Comment by kamilogorek** [comment]
> if it's just a session id set in the event context, that would be easy, but I believe it's not that.

That's exactly that 😄

https://github.com/getsentry/sentry-fullstory/blob/fda19a83ba5744d94522c6631f711037b84e5b23/src/SentryFullStory.ts#L33-L72

**Comment by marandaneto** [comment]
So technically one could just:

// Install fullstory on RN App
`yarn add @sentry/fullstory`

// Add SentryFullStory to the integrations list during Sentry SDK init
Sentry.init({
  dsn: '__DSN__',
  integrations: [ new SentryFullStory('__SENTRY_ORG_SLUG__'), ],
  // ...
});

@TannerLessen have you tried this?

**Comment by TannerLessen** [comment]
@marandaneto Yes I tried that originally as we make use of this integration in one of our React projects. However it fails due to a dependency on @fullstory/browser, which we do not use since we leverage the React Native FullStory SDK (@fullstory/react-native)

This feature request was to have a similar integration that functions with @fullstory/react-native as opposed to @fullstory/browser.. if feasible.

Let me know how to proceed if I've missed something!

**Comment by marandaneto** [comment]
Oh I see, that makes sense, we'd need an integration for `@fullstory/react-native`, indeed.

**Comment by nikolovlazar** [comment]
I did some comparison between the `react-native` and `browser` packages and here's what I came up with as a potential solution:
1. Create a new package (a clone of `@sentry/fullstory`) for the react-native specific implementation (`@sentry/fullstory-react-native`)
2. Replace the browser dependencies with the `react-native` ones
3. There are some minor refactoring that reflects the differences between the `browser` and `react-native` dependencies
   * `src/utils.ts`:
      - Import the `EventHint` type from `@sentry/types/types` instead
   * `src/types.ts`:
      - Remove the arg from `getCurrentSessionURL()` and set the return type to `Promise<string>`
   * `src/SentryFullStory.ts`:
      - Change both imports to `/react-native`
      - Import `FullStory` as a default instead of `* as FullStory`, since that's how it's defined in the `react-native` package
      - Remove the `Event` and `EventHint` import statement
      - Remove the types of `event` and `hint` from the `addGlobalEventProcessor`, let TypeScript infer them

Would it make more sense to create a completely new package named `'@sentry/fullstory-react-native'`, or create an integration script within the [react native integrations directory](https://github.com/getsentry/sentry-react-native/tree/main/src/js/integrations)?

**Disclaimer:** At the time of writing this comment I still haven't tested the actual functionality, but I'm curious of what you think is the right approach.
